File: baekjoon/DP/15489_파스칼삼각형.py
R, C, W = map(int, input().split())
arr = [[1, 1] for _ in range(R + W)]


def solution():
    for i in range(2, len(arr)):
        for j in range(2, i + 1):
            if j == i:
                arr[i].append(1)
            else:
                arr[i].append(arr[i - 1][j] + arr[i - 1][j - 1])
    sum = 0
    end = C + 1
    for i in range(R, R + W):
        for j in range(C, end):
            sum += arr[i][j]
        end += 1

    print(sum)


solution()

File: baekjoon/DP/15489_파스칼삼각형.js
const fs = require('fs')
const [R, C, W] = fs
  .readFileSync('../input.txt')
  .toString()
  .trim()
  .split(' ')
  .map(Number)

function solution() {
  let answer = 0
  const arr = Array.from(Array(R + W), () => [1, 1])

  for (let i = 2; i < arr.length; i++) {
    for (let j = 2; j <= i; j++) {
      arr[i].push((arr[i - 1][j] || 0) + arr[i - 1][j - 1])
    }
  }

  let end = C + 1
  for (let i = R; i < R + W; i++) {
    for (let j = C; j < end; j++) {
      answer += arr[i][j]
    }
    end += 1
  }

  return answer
}

console.log(solution())
